sumo_web3d/server/server.py

```diff
@@ -350,36 +356,49 @@ def start_sumo_executable(gui, sumo_args, sumocfg_file):
 def simulate_next_step():
     global last_lights, last_vehicles
     start_secs = time.time()
+    # Vehicles
     traci.simulationStep()
     end_sim_secs = time.time()
-    # Vehicles
+
     for veh_id in traci.simulation.getDepartedIDList():
-        # SUMO will not resubscribe to vehicles that are already subscribed, so this is safe.
+        # SUMO will not resubscribe to vehicles that
+        # are already subscribed, so this is safe.
         traci.vehicle.subscribe(veh_id, TRACI_VEHICLE_CONSTANTS)
-    vehicle_response = traci.vehicle.getSubscriptionResults().items()
-    # Vehicles are automatically unsubscribed upon arrival and deleted from vehicle list on next
+
+    # acquire the relevant vehicle information
+    ids = tuple(np.unique(traci.vehicle.getIDList() +
```

Comment:
I'd prefer not to introduce a dependence on numpy just for this.

You can make the IDs unique by putting them in a `set`.